tools/steam-shell.pike

```diff
@@ -458,25 +458,22 @@ int list(string what)
     write("Wrong usage\n");
     return 0;
   }
-  int flag=0;
   string toappend="";
   array(string) display = get_list(what);
-  string a="";
   if(sizeof(display)==0)
-    toappend = "There are no "+what+" in this room\n";
-  else
+    toappend = "There are no "+what+" in this room.\n";
+  else if (display[0]=="Invalid command")
+  {
+    write(display+"\n");
+  }
+  else{
     toappend = "Here is a list of all "+what+" in the current room\n";
+    write(toappend);
   foreach(display,string str)
   {
-    a=a+(str+"    ");
-    if(str=="Invalid command")
-    {
-      flag=1;
-      write(str+"\n");
-    }
+    write(sprintf("%-20s",str));
```

Comment:
Since the linux terminal is 80 char long , the column formation must be considered and the characters allocated to every object displayed.